clinica/pipelines/statistics_volume/statistics_volume_utils.py

```diff
@@ -45,31 +56,176 @@ def get_group_1_and_2(tsv, contrast):
+    If the values are categorical, then performs encoding of the categories.
+
+    Parameters
+    ----------
+    data: list of str
+        List of strings to be converted.
+
+    Returns
+    -------
+    list :
+        List of floats resulting from conversion.
+    """
     import numpy as np
+
+    from clinica.pipelines.statistics_volume.statistics_volume_utils import is_number
+
+    temp_data = [elem.replace(",", ".") for elem in data]
+    if all(is_number(elem) for elem in temp_data):
+        return [float(elem) for elem in temp_data]
+    # categorical variables (like Male; Female; M, F etc...)
+    unique_values = list(np.unique(np.array(data)))
+    return [unique_values.index(elem) for elem in data]
+
+
+def write_covariates(
+    covariates: list,
+    idx_group1: list,
+    idx_group2: list,
+    model: str,
+    covar: int,
+    covar_number: int,
+) -> None:
+    """Concatenate covariates for group1 and group2 and write them to disk.
+
+    Parameters
+    ----------
+    covariates: list
+        List of covariates to be concatenated and written.
+    idx_group1: list of int
+        List of indexes of first group
+    idx_group2: list of int
+        List of indexes of second group
+    model: str
+        Path to the matlab files with all the @TEXT replaced with the correct names
+    covar: int
+        Covariance
+    covar_number: int
+    """
+    from clinica.pipelines.statistics_volume.statistics_volume_utils import (
+        write_covariate_lines,
+    )
+
+    current_covar_data_group1 = [covariates[i] for i in idx_group1]
+    current_covar_data_group2 = [covariates[i] for i in idx_group2]
+    concatenated_covariates = current_covar_data_group1 + current_covar_data_group2
+    write_covariate_lines(model, covar_number, covar, concatenated_covariates)
+
+
+def write_matlab_model(
+    tsv: str,
+    contrast: str,
+    idx_group1: list,
+    idx_group2: list,
+    file_list: list,
+    template_file: str,
+):
+    """Create the matlab .m file for the instantiation of the 2-sample t-test model in SPM
+
+    Parameters
+    ----------
+    tsv: str
+        Path to the tsv file containing information on subjects/sessions with all covariates
+    contrast: str
+        Name of a column of the tsv
+    idx_group1: list of int
+        List of indexes of first group
+    idx_group2: list of int
+        List of indexes of second group
+    file_list: List
+        List of files used in the statistical test. Their order is the same as it appears on the tsv file
+    template_file: str
+        Path to the template file used to generate the .m file
+
+    Returns
+    -------
+    current_model: str
+        Path to the matlab files with all the @TEXT replaced with the correct names
+    covariates: list of str
+        Names of covariates
+    """
+    from numbers import Number
+    from os import remove
+    from os.path import abspath, isfile
```

Comment:
Same comment, `create_spm_output_folder` must be imported even if it is in the same module because Nipype is not able to find it.

```suggestion
    from os.path import abspath, isfile
    from clinica.pipelines.statistics_volume.statistics_volume_utils import create_spm_output_folder  # noqa
```

Reply:
Actually, you need to also import `set_output_and_groups`, `convert_to_numeric`, and `write_covariates` that you are also using in this function.